src/Consolonia.Core/Drawing/RenderTarget.cs

```diff
@@ -180,7 +183,9 @@ private void RenderToDevice()
                     _cache[x++, y] = Pixel.Empty;
             }
 
-            flushingBuffer.Flush();
+            _flushingBuffer.Flush();
+
+            _console.EndRender();
```

Comment:
We don't need to introduce one more buffer, we can fix existing one. The redundant calls to Flush can be made to depend on the console or something. Like for AnsiConsole we can skip flushing on Caret changes, for DefaultNetConsole we do flush on Caret moves etc.